aten/src/ATen/native/native_functions.yaml

```diff
@@ -6737,6 +6737,17 @@
   dispatch:
     CPU, CUDA: triangular_solve
 
+- func: linalg_solve_triangular.out(Tensor self, Tensor B, *, bool upper, bool left=True, bool unitriangular=False, Tensor(a!) out) -> Tensor(a!)
+  python_module: linalg
+  dispatch:
+    CPU, CUDA: linalg_solve_triangular_out
+
+- func: linalg_solve_triangular(Tensor self, Tensor B, *, bool upper, bool left=True, bool unitriangular=False) -> Tensor
```

Comment:
I don't think the API is "dirtier" when we make things optional, I just think it's clearer because it allows us to understand whether a kwarg has its default value or whether it was explicitly set by a user.

Reply:
Resolved to go with the `std::optional<bool>` API, but we'll add these as we add the compat arguments